openedx/core/lib/js_utils.py

```diff
@@ -81,3 +86,54 @@ def escape_js_string(js_string):
     js_string = decode.utf8(js_string)
     js_string = escapejs(js_string)
     return js_string
+
+
+def json(obj):
+    """
+    Mako filter that JSON dumps and escapes JSON that is safe to be embedded in HTML.
+
+    Usage:
+        Can be used inside a Mako template inside a <SCRIPT> as follows::
+        var my_json = ${my_object | n,json}
+
+        Use the "n" Mako filter above.  It is possible that the default filter
+        may include html escaping in the future, and this ensures proper escaping.
+
+        Ensure ascii in json.dumps (ensure_ascii=True) allows safe skipping of Mako's
+        default filter decode.utf8.
+
+    Arguments:
+        obj: The json object to be encoded and dumped to a string
+
+    Returns:
+        (string) Escaped encoded JSON
+
+    """
+    return escape_json_dumps(obj)
+
+
+# pylint: disable=invalid-name
+def js(js_string):
+    """
+    Mako filter that escapes a javascript string that is safe to be embedded in HTML.
+
+    Usage:
+        Can be used inside a Mako template inside a <SCRIPT> as follows::
+        var my_js_string = "${my_js_string) | n,js}"
+
+        Must include the surrounding quotes for the string.
+
+        Use the "n" Mako filter above.  It is possible that the default filter
+        may include html escaping in the future, and this ensures proper escaping.
+
+        Mako's default filter decode.utf8 is applied here since this default
+        filter is skipped in the Mako template with "n".
+
+    Arguments:
+        js_string (string): The javascript string to be escaped
```

Comment:
1) Please use sentence punctuation and capitalize things.
2) This is text, not a JavaScript string.